boards/nxp/frdm_ke17z512/doc/index.rst

```diff
@@ -0,0 +1,195 @@
+.. _frdm_ke17z512:
+
+NXP  FRDM-KE17Z512
+##################
+
+Overview
+********
+
+The FRDM-KE17Z512 is a development board for NXP Kinetis KE1xZ 32-bit
+MCU-based platforms. The onboard OpenSDAv2 serial and debug adapter,
+running an open source bootloader, offers options for serial
+communication, flash programming, and run-control debugging.
+
+.. figure:: frdm_ke17z512.webp
+   :align: center
+   :alt: FRDM-KE17Z512
+
+   FRDM-KE17Z512 (Credit: NXP)
+
+Hardware
+********
+
+- MKE17Z512VLL9 MCU (up to 96  MHz, 512 KB flash memory, 96 KB RAM,
+  and 100 Low profile Quad Flat Package (LQFP))
+- 3.3 V or 5 V MCU operation
+- 6-axis FXOS8700CQ digital accelerometer and magnetometer
+- RGB LED
+- Two user push-buttons
+- Thermistor
+- Arduino compatible I/O pin header
+- OpenSDA on-board debugger
+- Two Touch Electrodes
+
+For more information about the KE1xZ SoC and the FRDM-KE17Z512 board, see
+these NXP reference documents:
+
+- `KE1xZ Website`_
+- `KE1xZ Fact Sheet`_
+- `KE1xZ Reference Manual`_
+- `FRDM-KE17Z512 Website`_
+- `FRDM-KE17Z512 Quick Start Guide`_
+- `FRDM-KE17Z512 Reference Manual`_
+
+Supported Features
+==================
+
+The frdm_ke17z512 board configuration supports the following hardware
+features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| SYSTICK   | on-chip    | systick                             |
++-----------+------------+-------------------------------------+
+| PINCTRL   | on-chip    | pinctrl                             |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | uart                                |
```

Comment:
For consistency,  the driver/component section here on the UART
should be 
uart polling;
uart interrupt